judge/views/problem_data.py

```diff
@@ -44,7 +44,19 @@ class ProblemDataForm(ModelForm):
     def clean_zipfile(self):
         if hasattr(self, 'zip_valid') and not self.zip_valid:
             raise ValidationError(_('Your zip file is invalid!'))
-        return self.cleaned_data['zipfile']
+
+        zipfile = self.cleaned_data['zipfile']
+        if zipfile and zipfile.name == 'init.yml':
+            raise ValidationError(_('Zip files must not be named init.yml'))
+
+        return zipfile
+
+    def clean_generator(self):
```

Comment:
Dumb question: are we sure this is called?

Reply:
I did test it, so... I guess, yes?